src/crimson/os/seastore/segment_manager/block.cc

```diff
@@ -293,7 +293,15 @@ write_superblock(
     bufferptr(ceph::buffer::create_page_aligned(sb.block_size)),
     [=, &device](auto &bp)
   {
+    // Write the SEASTORE_SUPERBLOCK_SIGN at offset 0 before
+    // encoding the superblock. If the signature is added inside
+    // the block_sm_superblock_t structure (via DENC), it won't be
+    // written at offset 0 because DENC encoding add some metadata.
+    // Therefore prepend the signature to the bufferlist before
+    // encoding the superblock sothat signature always appears at the
+    // start of the written data.
```

Comment:
nit: This sections here would probably better fit in the commit message.
We can add a more specific comment here such as:
`// encode SEASTORE_SUPERBLOCK_SIGN at offset 0 before encoding anything else`